### ISSUE TITLE
MPIPool import error from emcee.utils

### ISSUE THREAD

**Issue by bikashd18** [dfm/emcee#343]
I was trying to run the emcee package for the MCMC algorithm in parallel mode using MPIPool. The line "from emcee.utils import MPIPool" is working on the python 2 environment but not in the python 3 environment. In python 3 environment it is showing an error: no module MPIPool exit. Can anyone please suggest me any idea to overcome this issue.

**Comment by dfm** [comment]
It's recommended that you use the [schwimmbad](https://schwimmbad.readthedocs.io/en/latest/) library. The tutorial is here: https://emcee.readthedocs.io/en/stable/tutorials/parallel/#mpi

**Comment by bikashd18** [comment]
Thanks, Dan!
The parallelization done; by both the methods using 'multiprocessing' and using 'Schwimmbad'.
I have some issues (I guess regarding 'Openmpi') with the lines:
"""
from schwimmbad import MPIPool
with MPIPool() as pool:
    if not pool.is_master():
        pool.wait()
        sys.exit(0)
    .....
    .....
"""
However, the problem has been solved by replacing the above lines by:
"""
from schwimmbad import MultiPool
with MultiPool() as pool:
    .......
    .......
"""
With the MPIPool, the error I got something like:
"""
[b:10259] mca: base: component_find: unable to open /usr/lib/openmpi/lib/openmpi/mca_shmem_mmap: /usr/lib/openmpi/lib/openmpi/mca_shmem_mmap.so: undefined symbol: opal_show_help (ignored)
[b:10259] mca: base: component_find: unable to open /usr/lib/openmpi/lib/openmpi/mca_shmem_posix: /usr/lib/openmpi/lib/openmpi/mca_shmem_posix.so: undefined symbol: opal_shmem_base_framework (ignored)
[b:10259] mca: base: component_find: unable to open /usr/lib/openmpi/lib/openmpi/mca_shmem_sysv: /usr/lib/openmpi/lib/openmpi/mca_shmem_sysv.so: undefined symbol: opal_show_help (ignored)
--------------------------------------------------------------------------
It looks like opal_init failed for some reason; your parallel process is
likely to abort.  There are many reasons that a parallel process can
fail during opal_init; some of which are due to configuration or
environment problems.  This failure appears to be an internal failure;
here's some additional information (which may only be relevant to an
Open MPI developer):

  opal_shmem_base_select failed
  --> Returned value -1 instead of OPAL_SUCCESS
--------------------------------------------------------------------------
--------------------------------------------------------------------------
It looks like orte_init failed for some reason; your parallel process is
likely to abort.  There are many reasons that a parallel process can
fail during orte_init; some of which are due to configuration or
environment problems.  This failure appears to be an internal failure;
here's some additional information (which may only be relevant to an
Open MPI developer):

  opal_init failed
  --> Returned value Error (-1) instead of ORTE_SUCCESS
--------------------------------------------------------------------------
--------------------------------------------------------------------------
It looks like MPI_INIT failed for some reason; your parallel process is
likely to abort.  There are many reasons that a parallel process can
fail during MPI_INIT; some of which are due to configuration or environment
problems.  This failure appears to be an internal failure; here's some
additional information (which may only be relevant to an Open MPI
developer):

  ompi_mpi_init: ompi_rte_init failed
  --> Returned "Error" (-1) instead of "Success" (0)
--------------------------------------------------------------------------
*** An error occurred in MPI_Init_thread
*** on a NULL communicator
*** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
***    and potentially your MPI job)
[b:10259] Local abort before MPI_INIT completed successfully; not able to aggregate error messages, and not able to guarantee that all other processes were killed!
"""
I think the error is related to opnmpi/mpi4py.
Any suggestions?
